toolchain/sem_ir/interface.h

```diff
@@ -20,6 +20,12 @@ struct Interface : public Printable<Interface> {
   // until we reach the `}` of the interface definition.
   auto is_defined() const -> bool { return defined; }
 
+  // Determines whether we're currently defining the interface. This is true
+  // between the braces of the interface.
+  auto is_being_defined() const -> bool {
```

Comment:
FWIW, I was considering "partially defined" for this state. But I was also mulling doing equivalent for other `definition_id.is_valid()` checks (class and fn), so maybe the `!defined` check is extra we don't generally want? *shrug*